staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -211,8 +211,21 @@ func (le *LeaderElector) renew(ctx context.Context) {
 		timeoutCtx, timeoutCancel := context.WithTimeout(ctx, le.config.RenewDeadline)
 		defer timeoutCancel()
 		err := wait.PollImmediateUntil(le.config.RetryPeriod, func() (bool, error) {
-			return le.tryAcquireOrRenew(), nil
+			var succeed bool
+			done := make(chan struct{})
+			go func() {
```

Comment:
This goroutines are going to leak. This concerns me.

Reply:
After setting timeout, the goroutine can definitely exit with at most less than twice `RenewDeadline` time.